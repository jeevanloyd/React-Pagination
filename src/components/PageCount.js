import React from "react";

export const PageCount = (props) => {
    return (

        <div key={props.limit} className="input-group mb-3" style={{ width: "auto" }}>
            <select
                className="form-select"
                id="pageselect"
                aria-label="Default select example"
                value={props.limit}
                onChange={(e) => props.setLimit(e.target.value)}
            >
                <option value="4">4 per Page</option>
                <option value="8" >
                    8 per Page
                </option>
                <option value="16">16 per Page</option>
                <option value="32">32 per Page</option>
                <option value="64">64 per Page</option>
            </select>
        </div>

    );
};
