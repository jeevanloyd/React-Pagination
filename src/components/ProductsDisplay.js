import React from "react";

 const ProductsDisplay = (props) => {
    return (
        <div key={props.data.id} className="col-md-3 col-12 d-flex justify-content-center">
            <div className="card w-100 mb-3">
                <img
                    src={props.data.product_image}
                    className="card-img-top w-50"
                    height={250}
                    style={{marginLeft:"25%", marginTop:"10px"}}
                    alt ="ProductsImage Display"
                />
                <div className="card-body">
                    <p className="card-text">
                        <span className="card-subtitle mb-2 h6 text-muted text-left">
                            {props.data.product_name}
                        </span> 
                    </p>
                    <p className="card-text">
                        <span className="card-subtitle mb-2 text-muted text-left">
                            {props.data.description}
                        </span>
                    </p>
                    <p className="card-text">
                        <span className="card-title text-left h6">
                            {props.data.price}{" "}
                        </span>
                    </p>
                </div>
            </div>
        </div>
    );
};

export default ProductsDisplay;