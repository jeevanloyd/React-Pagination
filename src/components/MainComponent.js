import React, { useEffect, useState } from "react";
import ReactPaginate from "react-paginate";
import { useDispatch } from "react-redux";
import PageCount from "./PageCount";
import ProductsDisplay from "./ProductsDisplay";


const MainComponent = () => {
    const [data, setdata] = useState([]);
    const [pageCount, setpageCount] = useState(0);
    const [limit, setLimit] = useState(8);
    const [total, setTotal] = useState(0);
    const dispatch = useDispatch()

    useEffect(() => {
        const getData = async () => {
            const res = await fetch(
                `http://localhost:4000/Data?_page=1&_limit=${limit}`
            );
            const data = await res.json();
            const total = res.headers.get("x-total-count");
            setTotal(total)
            setpageCount(Math.ceil(total / limit));
            setdata(data);
            dispatch({ type: "add_product", product: data })
        };

        getData();
    }, [dispatch, limit]);

    const fetchData = async (currentPage) => {
        const res = await fetch(
            `http://localhost:4000/Data?_page=${currentPage}&_limit=${limit}`
        );
        const data = await res.json();
        return data;
    };

    const handleClick = async (data) => {
        let currentPage = data.selected + 1;
        const dataRecieved = await fetchData(currentPage);
        setdata(dataRecieved)
        dispatch({ type: "add_product", product: data })
    };
    return (
        <>
            <div className="row d-flex justify-content-between mt-3">
                <div style={{ width: "auto" }}>{total} Products</div>
                < PageCount limit={limit} setLimit={setLimit} />
            </div>
            <hr />
            <div className="row"  >
                {data.map((item) => {
                    return (
                        <ProductsDisplay data={item} />
                    );
                })}
            </div>

            <hr />
            <ReactPaginate
                previousLabel={"previous"}
                nextLabel={"next"}
                breakLabel={"..."}
                pageCount={pageCount}
                marginPagesDisplayed={2}
                pageRangeDisplayed={2}
                onPageChange={handleClick}
                containerClassName={"pagination justify-content-end"}
                nextClassName={"page-item"}
                nextLinkClassName={"page-link"}
                pageLinkClassName={"page-link"}
                previousClassName={"page-item"}
                pageClassName={"page-item"}
                previousLinkClassName={"page-link"}
                breakClassName={"page-item"}
                breakLinkClassName={"page-link"}
                activeClassName={"active"}
            />
        </>
    );
};
export default MainComponent;