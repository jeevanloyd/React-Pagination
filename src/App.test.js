
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import HeaderComponent from './components/HeaderComponent';
import PageCount from './components/PageCount';
import ProductsDisplay from './components/ProductsDisplay';
import MainComponent from './components/MainComponent';
jest.mock('./components/ProductsDisplay', () => () => 'ProductsDisplay');
jest.mock('./components/MainComponent', () => () => 'MainComponent');

// Note: running cleanup afterEach is done automatically and cleanup DOM after the test is finished.
afterEach(cleanup);

describe('Render Header Component', () => {
  it('snapshot testing for Header component', () => {
    const component = renderer.create(
      <HeaderComponent />,
    );
    let Headertree = component.toJSON();
    expect(Headertree).toMatchSnapshot();
  })
})

describe('Render Pagecount Component', () => {
  it('snapshot testing for Pagecount component', () => {
    const component = renderer.create(
      <PageCount />,
    );
    let PagecountTree = component.toJSON();
    expect(PagecountTree).toMatchSnapshot();
  })
})

describe('Render Product Display Component', () => {
  let productData={
    Data: [
      {
          id: 1,
          price: "$87.68",
          product_name: "Amitriptyline Hydrochloride",
          description: "synergize efficient metrics",
          product_image: "http://dummyimage.com/307x328.bmp/ff4444/ffffff"
      },
    ]
  }
  it('snapshot testing for Product Display component', () => {
    const component = renderer.create(
      <ProductsDisplay props={productData}/>,
    );
    let ProductDisplayTree = component.toJSON();
    expect(ProductDisplayTree).toMatchSnapshot();
  })
})


describe('Render Main Component', () => {
  let productData={
    Data: [
      {
          id: 1,
          price: "$87.68",
          product_name: "Amitriptyline Hydrochloride",
          description: "synergize efficient metrics",
          product_image: "http://dummyimage.com/307x328.bmp/ff4444/ffffff"
      },
      {
        id: 2,
        price: "$58.53",
        product_name: "Zicam",
        description: "repurpose world-class metrics",
        product_image: "http://dummyimage.com/345x342.jpg/dddddd/000000"
    },
    ]
  }
  it('snapshot testing for main component', () => {
    const component = renderer.create(
      <MainComponent props={productData}/>,
    );
    let ProductDisplayTree = component.toJSON();
    expect(ProductDisplayTree).toMatchSnapshot();
  })
})





