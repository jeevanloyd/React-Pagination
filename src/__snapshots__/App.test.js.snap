// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Component snapshot testing for Header component 1`] = `
<div
  className="d-flex bg-dark text-white justify-content-center py-4"
>
  All Products
</div>
`;

exports[`Render Main Component snapshot testing for main component 1`] = `"MainComponent"`;

exports[`Render Pagecount Component snapshot testing for Pagecount component 1`] = `
<div
  className="input-group mb-3"
  style={
    Object {
      "width": "auto",
    }
  }
>
  <select
    aria-label="Default select example"
    className="form-select"
    id="pageselect"
    onChange={[Function]}
  >
    <option
      value="4"
    >
      4 per Page
    </option>
    <option
      value="8"
    >
      8 per Page
    </option>
    <option
      value="16"
    >
      16 per Page
    </option>
    <option
      value="32"
    >
      32 per Page
    </option>
    <option
      value="64"
    >
      64 per Page
    </option>
  </select>
</div>
`;

exports[`Render Product Display Component snapshot testing for Product Display component 1`] = `"ProductsDisplay"`;
